api.md

```diff
@@ -0,0 +1,73 @@
+
+* `/api/years/<year>/institutions`
+    * `GET` - Complete list of institutions a user can/could file for a given year. Structure: [institutions](https://github.com/cfpb/hmda-platform-ui/blob/c7590b1e6bd3243a8cd7909463ab8e2dc8c046a5/server/json/user1-institutions.json)
+
+* `/api/institutions/`
+    * `GET` - Complete list of institutions a user can/could file for *across all years*. Structure: [institutions](https://github.com/cfpb/hmda-platform-ui/blob/c7590b1e6bd3243a8cd7909463ab8e2dc8c046a5/server/json/user1-institutions.json), separated by year.
+
+* `/api/institutions/<institution>`
+    * `GET` - Data for a single institution *across all years*. Structure: [institution](https://github.com/cfpb/hmda-platform-ui/blob/c7590b1e6bd3243a8cd7909463ab8e2dc8c046a5/server/json/user1-institutions.json#L48-L56) collected in array based on id (`{institutions: [...]}`).
+
+* `/api/institutions/<institution>/years`
+    * `GET` - List of all years that a given institution filed/will file for. Structure: `{"years": ["2017"]}`
+  
+* `/api/years/<year>/institutions/<institution>`
+* `/api/institutions/<institution>/years/<year>`
+    * `GET` - Data for a single institution for a given year. Structure: [institutions](https://github.com/cfpb/hmda-platform-ui/blob/c7590b1e6bd3243a8cd7909463ab8e2dc8c046a5/server/json/user1-institutions.json#L48-L56)
+
+* `/api/years/<year>/institutions/<institution>/submissions`
+* `/api/institutions/<institution>/years/<year>/submissions`
+    * `GET` - List of all submissions for an institution in a given year. Structure: [submissions](https://github.com/cfpb/hmda-platform-ui/blob/c7590b1e6bd3243a8cd7909463ab8e2dc8c046a5/server/json/submissions.json#L26-L45) (without the extraneous key, which is just a convenience used by the dev server)
+    * `POST` - Create a new submission, incrementing the submission count. Return structure: `{"currentSubmission": 4}`
+
+* `/api/years/<year>/institutions/<institution>/submissions/<submission>`
+* `/api/institutions/<institution>/years/<year>/submissions/<submission>`
+    * `GET` - Data of a submission for an institution in a given year. Structure: [submissions](https://github.com/cfpb/hmda-platform-ui/blob/c7590b1e6bd3243a8cd7909463ab8e2dc8c046a5/server/json/submissions.json#L36-L45)
+    * `POST` - Endpoint for upload of LAR file. POST data: LAR file. Return structure: `{"progress": <progress endpoint>}`
+
+* `/api/years/<year>/institutions/<institution>/submissions/<submission>/progress`
+* `/api/institutions/<institution>/years/<year>/submissions/<submission>/progress`
+    * `GET` - Returns information about the progress of validation and submission in general. *This endpoint may support long-polling*. Structure:
+    ```
+    {
+      status: {
+        code: 3,
+        message: ""
+      },
```

Comment:
I _think_ the codes are from here:
https://github.com/cfpb/hmda-platform/issues/264#issuecomment-203686220

And the message would exist if the code is null. An example [here](https://github.com/cfpb/hmda-platform-ui/blob/master/server/json/submissions.json#L31). There are a few others in the file as well.

@wpears